src/openlifu/sim/sim_setup.py

```diff
@@ -100,12 +102,66 @@ def get_max_distance(self, arr: Transducer, units: Optional[str] = None):
 
     def get_size(self, dims: Optional[str]=None):
         dims = self.dims if dims is None else dims
-        n = [int(np.round(np.diff(ext)/self.spacing))+1 for ext in [self.x_extent, self.y_extent, self.z_extent]]
+        n = [int((np.round(np.diff(ext)/self.spacing)).item())+1 for ext in [self.x_extent, self.y_extent, self.z_extent]]
         return np.array([n[self.dims.index(dim)] for dim in dims]).squeeze()
 
     def get_spacing(self, units: Optional[str] = None):
         units = self.units if units is None else units
         return getunitconversion(self.units, units)*self.spacing
 
-    def transform_scene(self, scene, id: Optional[str] = None, name: Optional[str] = None, units: Optional[str] = None):
-        raise NotImplementedError
+    #TODO: since we don't have a concept of scene here, and that the simulation scene is needed in protocol.calc_solution,
+    # we will return each prepared object one-by-one.
+    #TODO: Missing the "markers" from matlab scene in "scene.transform(self, coords, matrix, options)"
+    #TODO: The arg transform needs to be added since transducer.matrix does not exists anymore
+    def setup_sim_scene(
```

Comment:
I prefer to have the segmentation in this function, we want to setup everything for the simulation here (sim grid with physical properties, and maybe some check to make sure transducer, target are in good units).
This will be completed with #150

Reply:
Okay, agreed :smile: